Ready for 2.* version of the iOS SDK

JKalash opened the pull request:
- Needs more testing to make sure it is 100% rock solid.
- Please let us know before you merge the request as we need to do a slight modification to our iOS docs.

Commit:
Ready for 2.* version of the SDK

Commit:
Ready for 2.* version of the SDK

avishayil commented:
Hi @JKalash, thanks for the pull request.

I revised the RCTOneSignal.xcodeproj file with some changes from errors I encountered trying to build.
1. Removed userdata from xcodeproj file.
2. Reverted User Search paths back to Header Search paths (causes compilation error).

Here's the link:
https://www.dropbox.com/s/6poe9d2zgvjo4b3/RCTOneSignal.xcodeproj.zip?dl=0

Please replace the file, Also, please update the Readme.md file of the repo with the updates to the API.
I'll continue to test this today, and will see if I have any more feedback.

Thanks!

avishayil commented:
Bug:

```
2016-08-21 12:03:42.651 Geektime[482:507984] *** Assertion failure in -[RCTEventEmitter sendEventWithName:body:](), /Users/avishay/geektime-app/node_modules/react-native/React/Modules/RCTEventEmitter.m:42
2016-08-21 12:03:42.657 [fatal][tid:com.facebook.react.RNOneSignalQueue] Exception 'bridge is not set. This is probably because you've explicitly synthesized the bridge in RCTOneSignal, even though it's inherited from RCTEventEmitter.' was thrown while invoking configure on target RNOneSignal with params (
)
```

Commit:
- Android 3.* support.
- Support for OSNotification / OSNotificationOpenResult objects.
- Added onNotificationsReceived event.
- Added syncHashedEmail & setLogLevel methods.
- Set SDK Type to "react".
- Updated README.
